frontend/src/TrendingTile.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from "react";
+import './UiTile.css';
+import './UiElements.css';
+import { EventBus, EventBusEventType } from "./EventBus";
+import { DeploymentManager } from "./DeploymentManager";
+import { ImageProvider } from "./ImageProvider";
+import { ToastManager } from "./ToastManager";
+import { LoginManager } from "./LoginManager";
+import { UserConfigurationDecoder } from "./UserConfigurationDecoder";
+import { SnippetManager } from "./SnippetManager";
+
+export interface TrendingTileState {
+    updating?: boolean
+    topics: TrendingTopic[]
+}
+export interface TrendingTileProps {
+    eventBus: EventBus
+    loginManager: LoginManager
+}
+
+export interface TrendingTopic {
+    name: string
+    tweet_volume: number
```

Comment:
I prefer camel case but this was dictated from the back end.

Reply:
The trending topics are fetched from the Twitter API and the resulting JSON is stored as-is into our elastic search backend.

Reply:
There is no transformation step after querying data from the elasticsearch via the Analytics API, since queries from the frontend are to be executed directly on the elasticsearch server (obiously permission-checked and only on the indices the executing user has access to) to enable a very dynamic front end dashboard experience.
The trend fetching endpoint is part of this Analytics API.

Reply:
Oh, okay. Makes sense.